drivers/input/i8042prt/pnp.c

```diff
@@ -263,6 +263,49 @@ i8042DetectMouse(
     INFO_(I8042PRT, "Mouse not detected\n");
 }
 
+static void
+i8042DetectMouseType(
+    IN PPORT_DEVICE_EXTENSION DeviceExtension)
+{
+    UCHAR Value;
+    /* Set the mouse to streaming mode, and disable error reporting (as per https://isdaman.com/alsos/hardware/mouse/ps2interface.htm) CORE-6901, CORE-12663, CORE-12434 */
+    i8042IsrWritePort(DeviceExtension, MOU_SET_STREAM_MODE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, MOU_NO_ERROR_REPORT, CTRL_WRITE_MOUSE);
+
+    /* Initialize Intellimouse or Intellimouse Explorer compatible mice (This fixes wheels) */
+    i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, 0xC8, CTRL_WRITE_MOUSE);
+
+    i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, 0x64, CTRL_WRITE_MOUSE);
+
+    i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, 0x50, CTRL_WRITE_MOUSE);
+
+    i8042IsrWritePort(DeviceExtension, MOU_READ_DEV_TYPE, CTRL_WRITE_MOUSE);
+    for(int i = 0; i < 10; i++) 
```

Comment:
```suggestion
    for (i = 0; i < 10; i++) 
```
Also a comment why the loop is done 10 times would be welcome.

Reply:
@BenNottelling selected this number in his PR.

Reply:
I simply picked it because it should loop through the mouse messages enough times to read everything.
Well, I kind assumed so